WireDomain/Tests/WireDomainTests/Synchronization/PushSupportedProtocolsUseCaseTests.swift

```diff
@@ -0,0 +1,357 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAPI
+import WireAPISupport
+import WireDataModel
+import WireDataModelSupport
+import XCTest
+
+@testable import WireDomain
+@testable import WireDomainSupport
+
+final class PushSupportedProtocolsUseCaseTests: XCTestCase {
+
+    private var sut: PushSupportedProtocolsUseCase!
+
+    private var coreDataStackHelper: CoreDataStackHelper!
+    private var stack: CoreDataStack!
+
+    private var context: NSManagedObjectContext {
+        stack.syncContext
+    }
+
+    // MARK: - Life cycle
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        coreDataStackHelper = CoreDataStackHelper()
+        stack = try await coreDataStackHelper.createStack()
+
+        sut = PushSupportedProtocolsUseCase(
+            featureConfigRepository: FeatureConfigRepository(
+                featureConfigsAPI: MockFeatureConfigsAPI(),
+                context: context
+            ),
+            userRepository: UserRepository(
+                context: context,
+                usersAPI: MockUsersAPI(),
+                selfUserAPI: MockSelfUserAPI()
+            )
+        )
+    }
+
+    override func tearDown() async throws {
+        sut = nil
+        stack = nil
+
+        try coreDataStackHelper.cleanupDirectory()
+        coreDataStackHelper = nil
+
+        try await super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func test_CalculateSupportedProtocols_AllActiveMLSClients_RemoteProteus() async throws {
+        // Given
```

Comment:
OPTIONAL: Consider using parameterized testing here where you define the test cases:

```swift
let testCases: [(migrationState:Scaffolding.MigrationState, supportedProtocols: Set<WireAPI.MessageProtocol>)] = [
   (migrationState: .disabled, supportedProtocols: [.proteus]),
   (migrationState: .notStarted, supportedProtocols: [.proteus]),
   (migrationState: .ongoing, supportedProtocols: [.proteus]),
   (migrationState: .finalised, supportedProtocols: [.proteus])
]
```

... then loop over them an assert the expected result. It _might be_ easier to read

Reply:
done thx